pkg/occlient/occlient.go

```diff
@@ -1370,22 +1379,93 @@ func (c *Client) CreateServiceInstance(componentName string, componentType strin
 				APIVersion: "servicecatalog.k8s.io/v1beta1",
 			},
 			ObjectMeta: metav1.ObjectMeta{
-				Finalizers: []string{"kubernetes-incubator/service-catalog"},
-				Name:       componentName,
-				Namespace:  c.namespace,
-				Labels:     labels,
+				Name:      componentName,
+				Namespace: c.namespace,
+				Labels:    labels,
 			},
 			Spec: scv1beta1.ServiceInstanceSpec{
 				PlanReference: scv1beta1.PlanReference{
 					ClusterServiceClassExternalName: componentType,
+					ClusterServicePlanExternalName:  servicePlan,
+				},
+				Parameters: util.BuildParameters(parameters),
+			},
+		})
+
+	if err != nil {
+		return errors.Wrapf(err, "unable to create the service instance %s for the service type %s and plan %s", componentName, componentType, servicePlan)
+	}
+
+	// Create the secret containing the parameters of the plan selected.
+	err = c.CreateSecret(c.namespace, componentName, parameters)
```

Comment:
Shouldn't this secret be deleted when ServiceInstance is deleted?
If we are creating it we should also clean it up when it is no longer needed.
It might make sense to set `metadata.ownerReferences` to point to the ServiceInstance, to make Kubenetes delete it when SI is deleted ( https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/)